tensorzero-core/src/providers/gcp_vertex_gemini/mod.rs

```diff
@@ -1868,6 +1905,198 @@ fn prepare_tools<'a>(
+                        function_response: GCPVertexGeminiFunctionResponse {
+                            name: Cow::Borrowed(&tool_result.name),
+                            response,
+                        },
+                    },
+                ));
+            }
+            Cow::Owned(ContentBlock::ToolResult(tool_result)) => {
+                let response = serde_json::json!({
+                    "name": tool_result.name,
+                    "content": tool_result.result
+                });
+
+                model_content_blocks.push(FlattenUnknown::Normal(
+                    GCPVertexGeminiContentPart::FunctionResponse {
+                        function_response: GCPVertexGeminiFunctionResponse {
+                            name: Cow::Owned(tool_result.name),
+                            response,
+                        },
+                    },
+                ));
+            }
+            Cow::Borrowed(ContentBlock::File(file)) => {
+                let FileWithPath {
+                    file,
+                    storage_path: _,
+                } = &**file;
+
+                model_content_blocks.push(FlattenUnknown::Normal(
+                    GCPVertexGeminiContentPart::InlineData {
+                        inline_data: GCPVertexInlineData {
+                            mime_type: file.mime_type.to_string(),
+                            data: Cow::Borrowed(file.data()?),
+                        },
+                    },
+                ));
+            }
+            Cow::Owned(ContentBlock::File(file)) => {
+                let FileWithPath {
+                    file,
+                    storage_path: _,
+                } = &*file;
+
+                model_content_blocks.push(FlattenUnknown::Normal(
+                    GCPVertexGeminiContentPart::InlineData {
+                        inline_data: GCPVertexInlineData {
+                            mime_type: file.mime_type.to_string(),
+                            data: Cow::Owned(file.data()?.to_string()), // Convert to owned String
+                        },
+                    },
+                ));
+            }
+            Cow::Borrowed(ContentBlock::Thought(ref thought))
+            | Cow::Owned(ContentBlock::Thought(ref thought)) => {
+                warn_discarded_thought_block(provider_type, thought);
+            }
+            Cow::Borrowed(ContentBlock::Unknown {
+                data,
+                model_provider_name: _,
+            }) => {
+                model_content_blocks.push(FlattenUnknown::Unknown(Cow::Borrowed(data)));
+            }
+            Cow::Owned(ContentBlock::Unknown {
+                data,
+                model_provider_name: _,
+            }) => {
+                model_content_blocks.push(FlattenUnknown::Unknown(Cow::Owned(data)));
+            }
+        }
+    }
+
+    // Check if we have any content blocks
+    if model_content_blocks.is_empty() {
```

Comment:
This should be removed - we might legitimately end up with no content blocks if they're all `Thought`s